rewrite-core/src/test/java/org/openrewrite/GitRemoteTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class GitRemoteTest {
+
+    @ParameterizedTest
+    @CsvSource(textBlock = """
+      https://github.com/org/repo, github.com, org/repo, org, repo
+      git@github.com:org/repo.git, github.com, org/repo, org, repo
+      ssh://github.com/org/repo.git, github.com, org/repo, org, repo
+                
+      https://gitlab.com/group/repo.git, gitlab.com, group/repo, group, repo
+      https://gitlab.com/group/subgroup/subergroup/subestgroup/repo.git, gitlab.com, group/subgroup/subergroup/subestgroup/repo, group/subgroup/subergroup/subestgroup, repo
+      git@gitlab.com:group/subgroup/subergroup/subestgroup/repo.git, gitlab.com, group/subgroup/subergroup/subestgroup/repo, group/subgroup/subergroup/subestgroup, repo
+      ssh://git@gitlab.com:22/group/subgroup/subergroup/subestgroup/repo.git, gitlab.com, group/subgroup/subergroup/subestgroup/repo, group/subgroup/subergroup/subestgroup, repo
+                
+      https://bitbucket.org/PRJ/repo, bitbucket.org, PRJ/repo, PRJ, repo
+      git@bitbucket.org:PRJ/repo.git, bitbucket.org, PRJ/repo, PRJ, repo
+      ssh://bitbucket.org/PRJ/repo.git, bitbucket.org, PRJ/repo, PRJ, repo
+                
+      https://dev.azure.com/org/project/_git/repo, dev.azure.com, org/project/repo, org/project, repo
+      git@ssh.dev.azure.com:v3/org/project/repo, dev.azure.com, org/project/repo, org/project, repo
+      ssh://ssh.dev.azure.com:22/v3/org/project/repo, dev.azure.com, org/project/repo, org/project, repo
```

Comment:
The following URLs fail to parse:
```
ssh://ssh.dev.azure.com:v3/org/project/repo
ssh://git@ssh.dev.azure.com:v3/org/project/repo
```

Reply:
Correct, do you think they should? To me it looks like a combination of scp style (: between host and path)and normal url (with scheme)

Reply:
I dunno. If you use the AD UI to copy a repo ssh URL, you'll get `git@ssh.dev.azure.com:v3/org/project/repo` but that's not a valid ssh URL. ssh URLs don't include paths but we're parsing URLs with paths. 

Something like `ssh://ssh.dev.azure.com:22/v3/org/project/repo` or ``ssh://git@ssh.dev.azure.com:22/v3/org/project/repo` wouldn't be returned by AD.

Personally, I think we only need to support the following forms:
```
https://dev.azure.com/org/project/_git/repo
https://user@dev.azure.com/org/project/_git/repo
git@ssh.dev.azure.com:v3/org/project/repo
```

Reply:
but we're not only accepting things that AD is returning. We accept valid cloneUrls, including the SCP style.

If a user builds their own url, it should at least be something valid (so either `ssh://user@host/path` or `user@host:path` but not `ssh://user@host:path`

Reply:
Sorry, I'm just commenting in the context of AD URLs...

I mentioned the following URLs failed to parse:
```
ssh://ssh.dev.azure.com:v3/org/project/repo
ssh://git@ssh.dev.azure.com:v3/org/project/repo
```

But after you asked if they _should_ be parseable, I would say no since these URLs would never be returned by AD. I'm not suggesting we change what we accept for other SCMs.

Reply:
ahh ok, yes, I misunderstood then.
So no change required, right?

Reply:
The only change I might suggest is removing `ssh://ssh.dev.azure.com:22/v3/org/project/repo` as a test input since it imples we officially support it. You can make that call...